### PR TITLE
topology improvements: layers, bounds

### PR DESCRIPTION
**Pull request by christianvogt**
**Fixes**
https://issues.redhat.com/browse/ODC-2187

This PR addresses two topology improvements:
- splitting bounds into separate position and dimension objects
- adding layers to the model so that it's not hard coded

The reason for splitting bounds into 2 objects is because of how mobx works for optimizing react renders. If a node only needs the height and width but uses bounds, then anytime the position changes, the node will need to be re-rendered. This happens a lot on drag move of a node. By switching to two separate objects. The parent ElementWrapper component will re-render when position changes and only the node will re-render when the dimensions change.

Layers was a simple fix by adding a `layers: string[]` array to the `Graph` model object. This initializes the layers and removes the hard coded layers to become the default in case no layers are provided.

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge

/assign @jeff-phillips-18

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/Topology.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ const graphModel: Model = {
     id: 'g1',
     type: 'graph',
     layout: COLA_LAYOUT,
+    layers: ['bottom', 'groups', 'groups2', 'default', 'top'],
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/src/components/topology/Topology.tsx:68]: Could use constants from topology for most.

File: frontend/packages/topology/stories/91-CollapsibleGroups.stories.tsx [hunk 2]
```diff
@@ -145,7 +145,7 @@ const TopologyViewComponent: React.FC<TopologyViewComponentProps> = ({ vis }) =>
             id="collapse-blue"
             label="Collapse Blue"
             isChecked={collapseBlue}
-            onChange={setCollapseBlue}
+            onChange={(checked) => setCollapseBlue(checked)}
```
Comment (jeff-phillips-18) [inline frontend/packages/topology/stories/91-CollapsibleGroups.stories.tsx:148]: why these changes?

Comment (christianvogt) [reply frontend/packages/topology/stories/91-CollapsibleGroups.stories.tsx:148]: Warnings were logged in the console because `onChange` actually has 2 params and the state functions only accept 1.
